server/webapp/WEB-INF/rails.new/app/presenters/api_v1/plugin_configuration_representer.rb

```diff
@@ -0,0 +1,30 @@
+##########################################################################
+# Copyright 2016 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  class PluginConfigurationRepresenter < BaseRepresenter
+    property :key
+    property :type, skip_nil: true
+    property :metadata, exec_context: :decorator
+
+    def metadata
+      represented.metadata.inject({}) do |hash, (k,v)|
```

Comment:
You seem to be transforming a `java.util.Map` returned by `PluginConfigurationViewModel#getMetadata`, any reason to have to do this? If you really needed a ruby hash (i presume for testing) you could have run `represented.metadata.to_hash` as described here https://github.com/jruby/jruby/wiki/CallingJavaFromJRuby#javautilmap-to-hash